### PR TITLE
BigQuery, MySQL, SQLite and SQL Server support

### PR DESCRIPTION
**Pull request by wendy-aw**
- Added a new script that translates the Postgres .sql files into the different dialects. Translation uses both sqlglot as well as regex operations.
- The script performs translation, and also creation of databases in the different db systems. It further checks that values have been inserted properly.
- Some modifications have been made to the original .sql files to strictly adhere to the data types set out in the DDL statements.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,14 +16,23 @@ export DBPORT=5432
...
 
+#### BigQuery, MySQL, SQLite, SQL Server
+To set up the data in these systems, you would need your credentials to be configured in `utils_dialects`. You can then run the following command to set up the databases:
+```
+python translate_ddl_dialect.py
+```
+This will create one new SQL file per database per dialect.
+For SQLite, the `.db` files will be saved in the folder `sqlite_dbs`.
+Note that BigQuery, MySQL and SQLite do not support schemas and hence the SQL files will be modified to skip schema creation.
+
```
Comment (rishsriv) [inline README.md:35]: Very helpful! Thanks for laying this down so clearly :D

File: defog_data/advising/advising.sql [hunk 2]
```diff
@@ -176,21 +176,21 @@ INSERT INTO public.comment_instructor (instructor_id, student_id, score, comment
 ;
 
 INSERT INTO public.course (course_id, name, department, number, credits, advisory_requirement, enforced_requirement, description, num_semesters, num_enrolled, has_discussion, has_lab, has_projects, has_exams, num_reviews, clarity_score, easiness_score, helpfulness_score) VALUES
-(1, 'Introduction to Computer Science', 'Computer Science', 'CS101', 3, NULL, NULL, 'This course introduces the basics of computer science.', 2, 2, true, false, true, false, 10, 5, 3, 4),
-(2, 'Advanced Calculus', 'Mathematics', 'MATH201', 4, 'CS101', NULL, 'This course covers advanced topics in calculus.', 1, 3, false, false, true, true, 5, 4, 2, 3),
-(3, 'Introduction to Physics', 'Physics', 'PHYS101', 3, NULL, 'MATH201', 'This course provides an introduction to physics principles.', 2, 1, true, true, true, true, 8, 4, 3, 5),
-(4, 'Distributed Databases', 'Computer Science', 'CS302', 3, NULL, 'CS101', 'This course provides an introduction to distributed databases.', 2, 2, true, true, false, true, 4, 2, 1, 5)
+(1, 'Introduction to Computer Science', 'Computer Science', 'CS101', '3', NULL, NULL, 'This course introduces the basics of computer science.', 2, 2, true, false, true, false, 10, 5, 3, 4),
```
Comment (rishsriv) [inline defog_data/advising/advising.sql:179]: Thanks for the fix here! TIL that Postgres likely automatically casted this to a string earlier

File: utils_dialects.py
Comment (rishsriv) [inline utils_dialects.py:1]: THANK YOU FOR THIS! Wow this would've taken forever to get right. Much appreciated!

**Review by rishsriv — APPROVED** [review]
Thanks for the fantastic work here Wendy!

File: defog_data/supplementary.py [hunk 3]
```diff
@@ -228,6 +228,19 @@ def load_embeddings(emb_path: str) -> tuple[dict, dict]:
             "journal.journalname,text,Name or title of the journal",
         ],
     },
+    "broker": {
```
Comment (wongjingping) [inline defog_data/supplementary.py:231]: Thanks for adding these placeholders!

File: utils_dialects.py
Comment (wongjingping) [reply utils_dialects.py:1]: +100 to this!

**Review by wongjingping — APPROVED** [review]

**Pull request merged**